rest_framework/fields.py

```diff
@@ -682,7 +682,7 @@ def to_internal_value(self, data):
             self.fail('max_string_length')
 
         try:
-            data = int(data)
+            data = int(re.compile(r'\.0*\s*$').sub('', str(data)))
```

Comment:
1. We should push this to `self.re_decimal` as per in Django so that it's only compiled once.
2. We should comment that this regex substitution is so that we allow eg '1.0' as an int, but not '1.2'